lib/util/stats.ts

```diff
@@ -105,6 +105,128 @@ export class PackageCacheStats {
   }
 }
 
+interface DatasourceCacheDataPoint {
+  datasource: string;
+  registryUrl: string;
+  packageName: string;
+  action: 'hit' | 'miss' | 'set' | 'skip';
+}
+
+export interface DatasourceCacheReport {
+  long: {
+    [datasource in string]: {
```

Comment:
looks ugly, can we use `Record`?

Reply:
It's ugly, but it better communicates the name/purpose of the particular level of nesting

Reply:
I still don't like that